test/chainrules.jl

```diff
@@ -275,7 +304,7 @@ end
             ZygoteRuleConfig(), my_namedtuple, 1., 2., 3.; rrule_f=rrule_via_ad
         )
         test_rrule(
-            ZygoteRuleConfig(), my_namedtuple, 1., (2.0, "str"), 3.; rrule_f=rrule_via_ad
+            ZygoteRuleConfig(), my_namedtuple, 1., (2.0, 2.4), 3.; rrule_f=rrule_via_ad
```

Comment:
Had to change this due to an issue in ChainRulesTestUtils
https://github.com/JuliaDiff/ChainRulesTestUtils.jl/issues/194

The Zygote and ChainRules code handles the string fine,
but ChainRulesTestUtils kinda freaks out about it.

Reply:
Maybe we can keep it as a `gradtest` then? I'm not super sure if CRTU is expected to be a dependency for every AD test suite.

Reply:
I am pretty sure `gradtest` can't handle it either -- gradtest can't handle namedtuples containing tuples.

> I'm not super sure if CRTU is expected to be a dependency for every AD test suite.

It is,  if you have `rrule_via_ad` overloaded. (or `rrule`)
Then you can use it to test all things.
It's a more robust version of `gradtest` that can handle more types, and doesn't e.g. get tricked by antisymmetric errors.

Reply:
Its good to have the coverage over different data types, but I guess its going to need a future PR. I can already sense some SMILES like application making use of strings outside of embeddings.